tools/clang/tools/dxcvalidator/dxcvalidator.cpp

```diff
@@ -191,32 +250,56 @@ uint32_t hlsl::validateWithDebug(
        OptDebugBitcode->Size >= UINT32_MAX))
     return E_INVALIDARG;
 
-  HRESULT hr = S_OK;
+  HRESULT HR = S_OK;
+  HRESULT ValidationStatus = S_OK;
   DxcThreadMalloc TM(DxcGetThreadMallocNoRef());
   try {
-    LLVMContext Ctx;
     CComPtr<AbstractMemoryStream> DiagMemStream;
-    hr = CreateMemoryStream(TM.GetInstalledAllocator(), &DiagMemStream);
-    if (FAILED(hr))
-      throw hlsl::Exception(hr);
-    raw_stream_ostream DiagStream(DiagMemStream);
-    llvm::DiagnosticPrinterRawOStream DiagPrinter(DiagStream);
-    PrintDiagnosticContext DiagContext(DiagPrinter);
-    Ctx.setDiagnosticHandler(PrintDiagnosticContext::PrintDiagnosticHandler,
-                             &DiagContext, true);
-    std::unique_ptr<llvm::Module> DebugModule;
-    if (OptDebugBitcode) {
-      hr = ValidateLoadModule((const char *)OptDebugBitcode->Ptr,
-                              (uint32_t)OptDebugBitcode->Size, DebugModule, Ctx,
-                              DiagStream, /*bLazyLoad*/ false);
-      if (FAILED(hr))
-        throw hlsl::Exception(hr);
+    CComPtr<IDxcBlob> HashedBlob;
+    HR = CreateMemoryStream(TM.GetInstalledAllocator(), &DiagMemStream);
+    if (FAILED(HR))
+      throw hlsl::Exception(HR);
+
+    // Run validation may throw, but that indicates an inability to validate,
+    // not that the validation failed (eg out of memory).
+    if (Flags & DxcValidatorFlags_RootSignatureOnly)
+      ValidationStatus =
+          runRootSignatureValidation(Shader, DiagMemStream, Flags, &HashedBlob);
+    else if ((Flags & DxcValidatorFlags_ModuleOnly) && IsInternalValidator)
```

Comment:
`DxcValidatorFlags_ModuleOnly` should allow validation of raw bitcode, and disable hashing.  `IsInternalValidator` should not be required, and I don't think this code handles this case properly.

Reply:
Current dxil.dll doesn't support DxcValidatorFlags_ModuleOnly, just return DXC_E_CONTAINER_INVALID.
Shall we change that in this PR?